abdm_integrator/crypto.py

```diff
@@ -0,0 +1,91 @@
+import hashlib
+from base64 import b64encode
+from dataclasses import asdict, dataclass
+from datetime import datetime, timedelta
+
+from fidelius import CryptoController, DecryptionRequest, EncryptionRequest, KeyMaterial
+
+CRYPTO_ALGORITHM = 'ECDH'
+CURVE = 'Curve25519'
+KEY_MATERIAL_EXPIRY = 60 * 60  # in seconds
+
+
+class ABDMCrypto:
+    """
+    Wrapper class to perform cryptography operations as per ABDM policy.
+    """
+
+    def __init__(self, key_material_dict=None, use_x509_for_transfer=False):
+        self.key_material = (ABDMKeyMaterial.from_dict(key_material_dict) if key_material_dict
+                             else self._generate_key_material())
+        self.transfer_material = self._get_transfer_material(use_x509_for_transfer)
+
+    @staticmethod
+    def _generate_key_material():
+        key_material = KeyMaterial.generate()
+        return ABDMKeyMaterial(
```

Comment:
Why create another class here? We can modify KeyMaterial itself we you need something more.

Reply:
Good point. Initially it was added as a wrapper class to easily work with fideliius key material.
As for more - it uses `frozen` to restrict modification, convenience methods to load from dict and export as dict. Will consider adding these at pyfidelius.

Reply:
Addressed in 51aa73bfddd7046773f6475e59d0d3911aa44857
Corresponding changes in PR; https://github.com/dimagi/pyfidelius/pull/2